File: my_android_v3/MyApplicationrecycle_view/app/src/main/java/com/example/myapplicationrecycle_view/Crypto/Base64Coder.java
package com.example.myapplicationrecycle_view.Crypto;


import org.bouncycastle.util.encoders.Base64;

public abstract class Base64Coder {

    public final static String ENCODING = "UTF-8";

    public static String encode(String data) throws Exception {

        byte[] b =  Base64.encode(data.getBytes(ENCODING));
        return new String(b, ENCODING);
    }

    public static String decode(String data) throws Exception {

        byte[] b = Base64.decode(data.getBytes(ENCODING));
        return new String(b, ENCODING);
    }

    public static byte[] decodeHex(String data) throws Exception {

        byte[] b = Base64.decode(data.getBytes(ENCODING));
        return b;
    }
}

File: RestfulAPI/crypto/Base64Coder.js
var encode = function(data){
    return Buffer.from(data).toString('base64');
}

var decode = function(data){
    return Buffer.from(data, 'base64').toString('utf-8');
}

var decodeHex = function (data) {
    return Buffer.from(data, 'base64')
}

module.exports = {
    encode: encode,
    decode: decode,
    decodeHex: decodeHex
}